src/vcastTestInterface.ts

```diff
@@ -353,27 +353,59 @@ export async function getResultFileForTest(testID: string) {
   // This function will return the path to the result file if it is already saved
   // in the globalTestStatus array, otherwise it will ask Python to generate the report
   let resultFile: string = globalTestStatusArray[testID].resultFilePath;
-  if (!fs.existsSync(resultFile)) {
-    let enviroPath = getEnviroPathFromID(testID);
 
+  // Check if the file already exists
+  if (!fs.existsSync(resultFile)) {
+    // Generate the environment path and request the test report from Python
+    const enviroPath = getEnviroPathFromID(testID);
     const commandStatus = await getTestExecutionReport(enviroPath, testID);
 
-    if (commandStatus.errorCode == 0) {
+    // Check if report generation was successful
+    if (commandStatus.errorCode === 0) {
       const firstLineOfOutput: string = commandStatus.stdout
         .split("\n", 1)[0]
         .trim();
-      resultFile = firstLineOfOutput.replace("REPORT:", "");
 
-      if (!fs.existsSync(resultFile)) {
+      // Handle the case where the output contains "REPORT"
+      if (firstLineOfOutput.includes("REPORT:")) {
+        // Verify if the generated report file actually exists
+        if (!fs.existsSync(resultFile)) {
+          vscode.window.showWarningMessage(
+            `The Report: ${resultFile} does not exist.`
+          );
+          vectorMessage(`The Report: ${resultFile} does not exist.`);
+        } else {
+          // This is the normal case --> delete the REPORT to only have the file name
+          resultFile = firstLineOfOutput.replace("REPORT:", "");
+        }
+      }
+      // If the first line of output contains "Error" --> Test result generation failed
+      else if (firstLineOfOutput.includes("Error:")) {
```

Comment:
So `commandStatus.errorCode === 0` is true even when we have a `Error:`?

Reply:
Yes because I think that the command execution perse does not fail 

https://github.com/vectorgrp/vector-vscode-vcast/blob/eaf7182ad9bccec5d4b78f5ca024133d3d6bb5a9/src/vcastCommandRunner.ts#L126-L131